black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/UserEndpoint.java

```diff
@@ -0,0 +1,62 @@
+package com.black.mulberry.rest.endpoint;
+
+import com.black.mulberry.core.entity.User;
+import com.black.mulberry.core.mapper.UserMapper;
+import com.black.mulberry.core.mapper.base.BaseMapper;
+import com.black.mulberry.core.repository.UserRepository;
+import com.black.mulberry.data.transfer.model.CreateUserDto;
+import com.black.mulberry.data.transfer.model.UserRole;
+import com.black.mulberry.rest.util.JwtTokenUtil;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@Slf4j
+public class UserEndpoint {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    private final BaseMapper baseMapper;
+
+    private final JwtTokenUtil jwtTokenUtil;
+
+    @PostMapping("/user")
+    public ResponseEntity<?> register(@RequestBody CreateUserDto createUserDto) {
+        Optional<User> existingUser = userRepository.findByEmail(createUserDto.getEmail());
+        if (existingUser.isPresent()) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).build();
+        }
+
+        User user = baseMapper.map(createUserDto);
+        user.setRole(UserRole.USER);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
no need to write logic here already exist that logic use them

Reply:
@sevakmartirosyan92  sonq sax tar service brat